tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,1628 @@
+    for the file with the given ID, C<Name> => name of the parameter and C<FullPath> => absolute
+    path of the file with the given ID.
+
+=cut
+sub getParameterFilesRef {
+    my($dbh, $tarchiveID, $dataDirBasePath, $scanTypesToDeleteRef) = @_;
+    
+    return [] if $tarchiveID eq 'NULL';
+
+    my $mriScanTypeJoin = @$scanTypesToDeleteRef
+        ? 'JOIN mri_scan_type mst ON (mst.ID=f.AcquisitionProtocolID) ' : '';
+    my $mriScanTypeAnd = @$scanTypesToDeleteRef
+        ? sprintf('AND mst.Scan_type IN (%s) ', join(',', ('?') x @$scanTypesToDeleteRef)) : '';
+    $query = 'SELECT FileID, Value, pt.Name FROM parameter_file pf '
+           . 'JOIN files f USING (FileID) '
+           . 'JOIN parameter_type AS pt USING (ParameterTypeID) '
+           . $mriScanTypeJoin
+           . "WHERE pt.Name IN ('check_pic_filename', 'check_nii_filename', 'check_bval_filename', 'check_bvec_filename') "
+           . 'AND ( '
+           . '         f.TarchiveSource = ? '
+           . '      OR f.SourceFileID IN (SELECT FileID FROM files WHERE TarchiveSource = ?) '
+           . ') '
+           . $mriScanTypeAnd;
+
+    my $filesRef = $dbh->selectall_arrayref(
+        $query, { Slice => {} }, $tarchiveID, $tarchiveID, @$scanTypesToDeleteRef
+    );
+    
+    # Set full path of every file
+    foreach(@$filesRef) {
+        my $fileBasePath = $_->{'Name'} eq 'check_pic_filename'
+            ? "$dataDirBasePath/" . PIC_SUBDIR : $dataDirBasePath;
+        $_->{'FullPath'} = $_->{'Value'} =~ /^\//
+            ? $_->{'Value'} : sprintf("%s/%s", $fileBasePath, $_->{'Value'});
+    }
+    
+    return $filesRef;
+}
+
+=pod
+
+=head3 getMriProtocolViolatedScansFilesRef($dbh, $tarchiveID, $dataDirBasePath, $scanTypesToDeleteRef)
+
+Get the absolute paths of all the files associated to a DICOM archive that are listed in 
+table C<mri_protocol_violated_scans>.
+
+INPUTS:
+  - $dbhr  : database handle reference.
+  - $tarchiveID: ID of the DICOM archive.
+  - $dataDirBasePath: config value of setting C<dataDirBasePath>.
+  - $scanTypesToDeleteRef: reference to the array that contains the list of scan type names to delete.
+
+RETURNS: 
+ - an array of hash references. Each hash has three keys: C<ID> => ID of the record in table
+   C<mri_protocol_violated_scans>, C<minc_location> => value of column C<minc_location> in table 
+   C<mri_protocol_violated_scans> for the MINC file found and C<FullPath> => absolute path of the MINC
+   file found.
+
+=cut
+sub getMriProtocolViolatedScansFilesRef {
+    my($dbh, $tarchiveID, $dataDirBasePath, $scanTypesToDeleteRef) = @_;
+    
+    # Return a reference to the enmpty array if:
+    # 1. The upload(s) do not have a tarchiveID, which means that no MINC file
+    #    were produced and consequently there cannot be any protocol violations
+    # OR
+    # 2. The -type option was used: only specific MINC files related to an archive
+    #    should be deleted and so we do not want to erase any protocol
```

Comment:
**that** should be deleted...

Reply:
No because then it would mean that the __archive__ is to be be deleted, which is not true.